metapool/controls.py

```diff
@@ -158,6 +159,38 @@ def get_controls_details_from_context(sample_context):
     return result
 
 
+def denormalize_controls_details(controls_details):
+    denormalized_controls_details = []
+    if controls_details is None:
+        return None
+
+    def _denormalize_details_record(a_record, qiita_id):
+        # remove the primary and secondary studies entries from the record,
+        # replace with a single QIITA_ID_KEY entry; leave everything else alone
+        denormalized_record = a_record.copy()
+        denormalized_record.pop(SECONDARY_STUDIES_KEY)
+        denormalized_record.pop(PRIMARY_STUDY_KEY)
+        denormalized_record[QIITA_ID_KEY] = qiita_id
+        return denormalized_record
+
+    for curr_record in controls_details.values():
+        for curr_secondary_study in curr_record[SECONDARY_STUDIES_KEY]:
+            curr_denorm_secondary_record = _denormalize_details_record(
+                curr_record, curr_secondary_study)
+            denormalized_controls_details.append(curr_denorm_secondary_record)
+        # next secondary study, if any
+
+        curr_denorm_primary_record = _denormalize_details_record(
+            curr_record, curr_record[PRIMARY_STUDY_KEY])
+        denormalized_controls_details.append(curr_denorm_primary_record)
+    # next controls_detail record
+
+    denormalized_controls_details = sorted(
+        denormalized_controls_details,
+        key=lambda k: (k[SAMPLE_NAME_KEY], k[QIITA_ID_KEY]))
```

Comment:
I would anticipate sorting by study then sample? Or is the result more interpretable on a sample order basis?

Reply:
Well, the place the output is used is in `mg-scripts` where it is making the sample info files for the blanks, which is sample-level work, which is why I think this made sense to me.  I actually only put a sort in here to make it stable for my unit tests, so I suppose we could change it so something else :)

Reply:
Oh I see. Okay, makes sense. If the sort order doesn't actually matter, then I agree that having _a_ sort is valuable but I dont' think it matters if sample or study takes precedence